internal/app/mistapiconnector/mistapiconnector_app.go

```diff
@@ -122,14 +129,12 @@ func NewMac(mistHost string, mapi *mist.API, lapi *livepeer.API, balancerHost st
 			err = fmt.Errorf("mist-api-connector: Error connecting ETCD err=%w", err)
 			return nil, err
 		}
-		sess, err = concurrency.NewSession(cli, concurrency.WithTTL(etcdSessionTTL))
+		sess, err = newEtcdSession(cli)
```

Comment:
Just realized we can also force a `cli.Sync()` immediately here, so we don't have to wait for the auto-sync interval before the client picks all cluster endpoints. That only if we still might take some time until having the full member list in the endpoints configuration.

That said, also came to my mind that the session might be lost sometimes because we currently only _create_ by the time we have a single endpoints in the `etcd` client? Than the 3 endpoints could be the thing that fixes the session issues, but it still feels like something that could happen anytime since the session is bound to a manual renewal of a time-limited lease. Also not sure if the `etcd` protocol requires that you send the writes to all members of the cluster for it to really have an effect, sounds like something the `etcd` servers would handle in between themselves.

Reply:
> Just realized we can also force a cli.Sync() immediately here,

Let's do it.

>  Also not sure if the etcd protocol requires that you send the writes to all members

No, client sends to one server, and then servers handle that between themselves.